livekit/src/room/participant/local_participant.rs

```diff
@@ -544,4 +642,205 @@ impl LocalParticipant {
     pub fn kind(&self) -> ParticipantKind {
         self.inner.info.read().kind
     }
+
+    pub async fn perform_rpc(
+        &self,
+        destination_identity: String,
+        method: String,
+        payload: String,
+        response_timeout_ms: Option<u32>,
+    ) -> Result<String, RpcError> {
+        let response_timeout = Duration::from_millis(response_timeout_ms.unwrap_or(10000) as u64);
+        let max_round_trip_latency = Duration::from_millis(2000);
+
+        if payload.len() > MAX_PAYLOAD_BYTES {
+            return Err(RpcError::built_in(RpcErrorCode::RequestPayloadTooLarge, None));
+        }
+
+        if let Some(server_info) = self.inner.rtc_engine.get_latest_join_response().server_info {
```

Comment:
ah ok, so you should use the one inside signal_client, there is already the last join response there

Reply:
https://github.com/livekit/rust-sdks/blob/3de6fa60f3cdafc2c106a7f370724eddf1e92fdc/livekit-api/src/signal_client/mod.rs#L164

Reply:
oh yeah perfect - i was looking for something called "latest_join_response" like in JS so i missed that. i can access it here as `self.inner.rtc_engine.session().signal_client().join_response()`